### PR TITLE
chore: refactor k8s pkg

### LINKED ISSUE #111: refactor `k8s` package to improve code quality
It's a mess currently..

### PR DESCRIPTION
**Pull request by mojtaba-esk**
This PR proposes a refactor change on k8s pkg. It is attempted to keep the rest of the code intact unless it breaks with the new changes.
This is the first round, after finishing the refactor for the entire project we might need to go through the code from A to Z a couple of times to improve it even further.
Closes #111

### CODE REVIEW

File: pkg/k8s/k8s.go [hunk 1]
```diff
@@ -2,81 +2,83 @@
...
-		return false
-	}
-	if _, err := os.Stat(certPath); err != nil {
-		return false
-	}
-
-	return true
+	_, errToken := os.Stat(tokenPath)
+	_, errCert := os.Stat(certPath)
+	return errToken == nil && errCert == nil
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply pkg/k8s/k8s.go:68]: i think it is not very idiomatic, but i like it too ;)

File: pkg/k8s/k8s_configmap.go [hunk 2]
```diff
@@ -4,109 +4,77 @@ import (
...
+	exists, err := c.ConfigMapExists(ctx, name)
 	if err != nil {
 		return nil, err
 	}
 	if exists {
 		return nil, ErrConfigmapAlreadyExists.WithParams(name)
 	}
 
-	cm, err := prepareConfigMap(namespace, name, labels, data)
+	cm, err := prepareConfigMap(c.namespace, name, labels, data)
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply pkg/k8s/k8s_configmap.go:45]: exactly. there will be only one place that user has to pass namespace

File: pkg/knuu/instance_helper.go [hunk 3]
```diff
@@ -69,16 +69,17 @@ func (i *Instance) getLabels() map[string]string {
 		"knuu.sh/scope":                k8s.SanitizeName(testScope),
 		"knuu.sh/test-started":         startTime,
 		"knuu.sh/name":                 i.name,
-		"knuu.sh/k8s-name":             i.k8sName,
+		"knuu.sh/k8sClient-name":       i.k8sName,
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply pkg/knuu/instance_helper.go:72]: oops that's a typo, will fix it asap

[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply pkg/knuu/instance_helper.go:72]: nice catch btw ;)

File: pkg/k8s/k8s_pvc.go [hunk 4]
```diff
@@ -9,17 +9,40 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+func (c *Client) DeployPersistentVolumeClaim(ctx context.Context, name string, labels map[string]string, size resource.Quantity) error {
```
Comment (smuu) [inline pkg/k8s/k8s_pvc.go:12]: We use `Create`, `Get`, and `Delete` for other resources. Should we use the same function naming for all resources to be more consistent?

File: pkg/k8s/k8s_replicaset.go [hunk 5]
```diff
@@ -12,62 +12,36 @@ import (
...
+	Namespace string            // Namespace of the ReplicaSet
+	Labels    map[string]string // Labels to apply to the ReplicaSet, key/value represents the name/value of the label
+	Replicas  int32             // Replicas is the number of replicas
+	PodConfig PodConfig         // PodConfig represents the pod configuration
 }
 
-// DeployReplicaSet creates a new replicaSet in the given namespace if it doesn't already exist.
-func DeployReplicaSet(ctx context.Context, replicaSetConfig ReplicaSetConfig, init bool) (*appv1.ReplicaSet, error) {
+// DeployReplicaSet creates a new replicaSet in namespace that k8s is initialized with if it doesn't already exist.
+func (c *Client) DeployReplicaSet(ctx context.Context, rsConfig ReplicaSetConfig, init bool) (*appv1.ReplicaSet, error) {
```
Comment (smuu) [inline pkg/k8s/k8s_replicaset.go:24]: We use `Create`, `Get`, and `Delete` for other resources. Should we use the same function naming for all resources to be more consistent?

File: pkg/k8s/k8s_service.go [hunk 6]
```diff
@@ -10,110 +10,80 @@ import (
...
 	if err != nil {
 		return nil, ErrGettingService.WithParams(name).Wrap(err)
 	}
 	return svc, nil
 }
 
-// DeployService deploys a service if it does not exist.
-func DeployService(
-	namespace,
+func (c *Client) DeployService(
```
Comment (smuu) [inline pkg/k8s/k8s_service.go:21]: We use `Create`, `Get`, and `Delete` for other resources. Should we use the same function naming for all resources to be more consistent?

File: pkg/knuu/instance_helper.go [hunk 7]
```diff
@@ -180,17 +175,17 @@ func (i *Instance) destroyPod() error {
 }
 
 // deployService deploys the service for the instance
-func (i *Instance) deployOrPatchService() error {
+func (i *Instance) deployOrPatchService(ctx context.Context) error {
```
Comment (smuu) [inline pkg/knuu/instance_helper.go:178]: We use `Create`, `Get`, and `Delete` for other resources. Should we use the same function naming for all resources to be more consistent?

**Review by smuu — CHANGES_REQUESTED** [review]
This looks amazing :rocket:

Thank you for the time spent on this.

- [ ] Just added some small comments regarding function naming
- [ ] I assume you tested it with those changes?
- [ ] It does not look like that, but do those changes break anything in the `knuu` packages that need changes in the tests themselves?

File: pkg/knuu/instance_helper.go (hunk 7 above)
Comment (mojtaba-esk) [reply pkg/knuu/instance_helper.go:178]: now it is renamed for k8s pkg, to keep things separate. I suggest to take care of this for knuu pkg in a separate PR. I am creating an issue for it.

**Review by smuu — APPROVED** [review]
:rocket:

**Pull request merged**